### PR TITLE
Madison/fix class counts

### PR DESCRIPTION
**Pull request by madisonmay**

### CODE REVIEW

File: finetune/nn/target_blocks.py [hunk 1]
```diff
@@ -321,11 +321,12 @@ def seq_lab_internal(hidden):
...
 
         log_likelihood = 0.0
-        default_lengths = kwargs.get('max_length') * tf.ones(tf.shape(hidden)[0], dtype=tf.int64)
+        
+        default_lengths = kwargs.get('max_length') * tf.ones(tf.shape(hidden)[0], dtype=tf.int32)
         if pool_idx is None:
             pool_idx = default_lengths
         else:
-            pool_idx = tf.where(tf.equal(pool_idx, 0), default_lengths, pool_idx)
+            pool_idx = tf.where(tf.equal(pool_idx, 0), default_lengths, tf.cast(pool_idx, dtype=tf.int32))
```
Comment (madisonmay) [inline finetune/nn/target_blocks.py]: Snuck this change in here -- I think something about this code was env dependent

File: finetune/encoding/target_encoders.py [hunk 2]
```diff
@@ -66,6 +66,21 @@ def transform(self, y):
         labels = super().transform(y)
         return self._make_one_hot(labels)
 
+    def inverse_transform(self, one_hot):
+        if len(one_hot.shape) == 2:
```
Comment (benleetownsend) [inline finetune/encoding/target_encoders.py]: What does it normally equal? Is it not always `[batch, num_classes]`?

**Comment by benleetownsend** [comment]
Nicely done!! Can we check what the behaviour is if you run this on other types other than classification and sequence labelling. I suspect it will either silently fail and not re-weight or fail with a traceback that doesn't make sense. 

As a wider design thing. We need to think about how we create a general way to detect and warn when things are incompatible. As finetune grows this will just be more and more impractical to manage on a case by case basis.

File: finetune/encoding/target_encoders.py (hunk 2 above)
Comment (madisonmay) [reply finetune/encoding/target_encoders.py]: So... the inverse_transform() call of this function is also unfortunately used to convert from class ids to class names in predict() right now. Which violates the assertion that `x = inverse_transform(transform(x))`

Comment (madisonmay) [reply finetune/encoding/target_encoders.py]: Maybe the best solution to this is to ensure that the output of the predict() method of any model type that uses OneHotLabelEncoder as it's target encoder is one hot encoded

**Pull request merged**